tensorboard/defs/vulcanize.bzl

```diff
@@ -73,7 +73,10 @@ def _tensorboard_html_binary(ctx):
   # webfiles manifest
   manifest_srcs = [struct(path=ctx.outputs.html.path,
                           longpath=long_path(ctx, ctx.outputs.html),
-                          webpath=ctx.attr.output_path)]
+                          webpath=ctx.attr.output_path),
+                   struct(path=ctx.outputs.js.path,
+                          longpath=long_path(ctx, ctx.outputs.js),
+                          webpath=ctx.attr.js_path)]
```

Comment:
The `js_path` attribute is optional, and the following targets don’t
provide it:

```
//tensorboard/components/vz_line_chart2/microbenchmark:binary
//tensorboard/components:polymer_lib_binary
//tensorboard/components_polymer3/vz_line_chart2/microbenchmark:binary
//tensorboard/components_polymer3:polymer_lib_binary
//tensorboard/plugins/graph/polymer3/tf_graph_app:binary
//tensorboard/plugins/graph/tf_graph_app:binary
//tensorboard/plugins/projector/vz_projector:devserver
```

So it looks like this adds a struct with `webpath = ""`; curious how/why
that works? (And it does: the `vz_projector:devserver` still seems to
work fine for me.)

Reply:
You are right. I, too, am surprised that empty webpath is treated correctly.

I added some checks before adding the manifest.